src/main/java/ch/njol/skript/entity/ThrownPotionData.java

```diff
@@ -46,34 +47,33 @@
 		EntityData.register(ThrownPotionData.class, "thrown potion", ThrownPotion.class, "thrown potion");
 	}
 	
-	private final static Adjective m_adjective = new Adjective("entities.thrown potion.adjective");
+	private static final Adjective m_adjective = new Adjective("entities.thrown potion.adjective");
+	private static final boolean RUNNING_LEGACY = !Skript.isRunningMinecraft(1, 13);
 	
 	@Nullable
 	private ItemType[] types;
 	
 	@Override
 	protected boolean init(final Literal<?>[] exprs, final int matchedPattern, final ParseResult parseResult) {
 		if (exprs.length > 0 && exprs[0] != null) {
-			if ((Converters.convert((ItemType[]) exprs[0].getAll(), ItemType.class, new Converter<ItemType, ItemType>() {
-				@Override
-				@Nullable
-				public ItemType convert(final ItemType t) {
-					ItemType r = null;
-					for (final ItemData d : t.getTypes()) {
-						if (d.getType() == Material.POTION) {
-							if (r == null)
-								r = new ItemType(d);
-							else
-								r.add(d);
-						}
-					}
-					return r;
+			return (types = Converters.convert((ItemType[]) exprs[0].getAll(), ItemType.class, t -> {
+				Material type = t.getMaterial();
+				// If the itemtype is a potion, lets make it a splash potion (required by Bukkit)
+				// Due to an issue with 1.12.2 and below, we have to force a lingering potion to be a splash potion
+				if (type == Material.POTION || (type == Material.LINGERING_POTION && RUNNING_LEGACY)) {

```

Comment:
Don't do Material checks, put the corresponding alias in a constant through Aliases#javaItemtype instead.

Reply:
Ah yes you're right, totally forgot about that thanks. Ill switch them up.

Reply:
Ok these have now been changed!